src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java

```diff
@@ -279,49 +279,70 @@ public static void scan(Run build, PrintStream scanLog) {
+                bufBuildFailCause.append("*Failure Name:* ");
+                bufBuildFailCause.append(foundCause.getName());
+                bufBuildFailCause.append("\n");
+                bufBuildFailCause.append("*Failure Categories:* ");
+                bufBuildFailCause.append(foundCause.getCategories());
+                bufBuildFailCause.append("\n");
+                bufBuildFailCause.append("*Description:* ");
+                bufBuildFailCause.append(foundCause.getDescription());
+            } else {
+                // Only notify the selected categories even if others occur
                 List<String> categories = foundCause.getCategories();
                 if (categories != null) {
                     for (String category : categories) {
                         if (failureCategoryMatches(category, slackFailureCauseCategories)) {
                             notifySlackOfFailure = true;
-                            break;
+                            //Add two new lines between found categoreis
+                            if (bufBuildFailCause.length() != 0) {
+                                bufBuildFailCause.append("\n\n");
+                            }
+                            // Create list for slack message with failure Name, Cegory and Description from build
+                            bufBuildFailCause.append("*Failure Name:* ");
+                            bufBuildFailCause.append(foundCause.getName());
+                            bufBuildFailCause.append("\n");
+                            bufBuildFailCause.append("*Failure Categories:* ");
+                            bufBuildFailCause.append(foundCause.getCategories());
+                            bufBuildFailCause.append("\n");
+                            bufBuildFailCause.append("*Description:* ");
+                            bufBuildFailCause.append(foundCause.getDescription());
+
                         }
                     }
                 }
             }
-            /* Create list for slack message with failure causes from build */
-            if (bufBuildFailCause.length() == 0) {
-                bufBuildFailCause.append(foundCause.getName());
-            } else {
-                bufBuildFailCause.append("\n");
-                bufBuildFailCause.append(foundCause.getName());
-            }
         }
 
         /* Notify slack if configured to report "ALL" failures or if one of the failure categories matched. */
         if (notifySlackOfAllFailures || notifySlackOfFailure) {
             SlackMessageProvider slack = new SlackMessageProvider();
 
             StringBuilder s = new StringBuilder("Job *\"" + buildName + "\"*");
-            s.append(" build *#" + buildNum + "* FAILED due to following failure causes: \n```");
-            s.append(bufBuildFailCause.toString() + "```\nSee ");
+            s.append(" build *#" + buildNum + "* FAILED due to following failure causes: \n");
```

Comment:
Using StringBuilder to optimize string concatenations and then doing contcatenations :)
```suggestion
            s.append(" build *#").append(buildNum).append("* FAILED due to following failure causes: \n");
```
You should fix the constructor also.